### PR TITLE
CLI: handle directories passed with -p

### PR DESCRIPTION
**Pull request by ajafff**
#### PR checklist

- [x] Addresses an existing issue: #2692 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:
Search for tsconfig.json in specified directory. Exit with error if  there is none.
Ref: #2692
[enhancement] cli: `-p` option handles directories

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -146,11 +146,14 @@ export class Runner {
         let program: ts.Program | undefined;
 
         if (this.options.project != null) {
-            if (!fs.existsSync(this.options.project)) {
+            let project: string;
+            try {
+                project = findTsconfig(this.options.project);
```
Comment (andy-hanson) [inline src/runner.ts]: I think this function should handle error handling itself (and return `string | undefined` instead of throwing an error that gets ignored.

File: src/runner.ts [hunk 2]
```diff
@@ -175,9 +178,6 @@ export class Runner {
                 // if not type checking, we don't need to pass in a program object
                 program = undefined;
             }
-        } else if (this.options.typeCheck) {
```
Comment (andy-hanson) [inline src/runner.ts]: Why did you move this code?

Comment (ajafff) [inline src/runner.ts]: all static option validation is done in tslint-cli.ts
runner.ts should only be responsible to catch errors that occur at runtime

File: src/runner.ts (hunk 1 above)
Comment (ajafff) [inline src/runner.ts]: done
